lib/matplotlib/axis.py

```diff
@@ -1710,6 +1710,41 @@ def get_tick_space(self):
         # Must be overridden in the subclass
         raise NotImplementedError()
 
+    def _get_ticks_position(self):
+        """
+        Helper for `XAxis.get_ticks_positions` and `YAxis.get_ticks_position`.
+
+        Check the visibility of tick1line, label1, tick2line, and label2 on
+        the first major and the first minor ticks, and return
+
+        - 1 if only tick1line and label1 are visible;
+        - 2 if only tick2line and label2 are visible;
```

Comment:
comment here why these are left ambiguous....

Reply:
what do you mean?

Reply:
Why they don’t return strings.

Reply:
Because they become "top"/"bottom" for the x-axis but "left"/"right" for the y-axis?  For an abstract Axis, 1 and 2 *are* the names of the ticks and labels, they don't have string names yet.

Reply:
Yes, I understand, but it took me a bit, and it will be even more mysterious for someone who doesn't have the diff to look at.  Just adding a one-line comment here would make it clearer...

Reply:
I think it would be useful to clarify that 1 means "top" or "bottom" and 2 means "left" or "right" in the docstring here.